### ISSUE TITLE
Increased leniency on the start of the shellcheck directive format

### ISSUE THREAD

**Issue by halostatue** [koalaman/shellcheck#2686]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I’ve recently started using [Bashly](https://bashly.dannyb.co) to tame the construction of larger shell scripts, and it has two forms of comments in the script partials that it generates:

```bash
## These will be stripped from the partial before merging into the final script.
# These will be passed to the final script.
```

I want to be able to put `shellcheck` directives in one of the stripped comments:

```bash
## shellcheck disable=SC2034
declare -gA some_values
```

However, it appears that at least with `shellcheck 0.9.0`, that is not an accepted directive format and I am still warned about the unused value.
